### DISCUSSION TITLE
直播信息扩充组件看起来是可以改进的

### DISCUSSION THREAD

**Discussion by caikun233** [the1812/Bilibili-Evolved#4908] [功能建议 / Ideas]
先说一下我完全没看代码（）
看起来它现在是像这样运行的
```
page = 1
while True:
    请求 https://api.live.bilibili.com/xxxxxxx/GetWebList?page={page}
    if len([data].[rooms]) < 9
        break
    page += 1
```
挺麻烦的就是说，我刚发现个API是这样的（直播站右上角，鼠标移到关注按钮上的弹出窗用的就是它）
```
GET https://api.live.bilibili.com/xlive/web-ucenter/v1/xfetter/GetWebList?hit_ab=true&_={现在的毫秒级时间戳}
```
cookie还是照常带bili_jct和SESSDATA，header里面带个Host就足够

这个API会一股脑把该账号关注且正在直播的房间吐出来，响应的信息不差啥，就缺些online, live_time, area, 以及cover_from_user等这个功能用不到的东西，并且都有响应只不过是空值或0，而需要的face, uname, title, room_id这些是一点不缺

所以似乎可以只请求一次这个API，可以略微的减少那么一丁点占用吧大概（）

因为比较无关紧要所以没开issue

**Comment by oxygenkun** [comment]
#4964 was merged. This discussion can be marked as `complete`.
